waspc/cli/src/Wasp/Cli/Command/Build.hs

```diff
@@ -121,13 +123,15 @@ build = do
       -- extends from it.
       liftIO $
         copyFile
-          (waspProjectDir </> srcTsConfigInWaspLangProject)
+          (waspProjectDir </> srcTsConfigPath)
           tsconfigJsonInBuildDir
 
       -- A hacky quick fix for https://github.com/wasp-lang/wasp/issues/2368
       -- We should remove this code once we implement a proper solution.
       ExceptT $ updateJsonFile removeWaspConfigFromDevDependenciesArray packageJsonInBuildDir
       ExceptT $ updateJsonFile removeAllMentionsOfWaspConfigInPackageLockJson packageLockJsonInBuildDir
+      where
+        srcTsConfigPath = AS.srcTsConfigPath appSpec
```

Comment:
We should always avoid nested where blocks, they're a common Haskell antipattern.

The reason is that they make the control flow difficult to follow and allow too much mixing of monadic and non-nomadic code. In short, whenever there's a nested `where`, the code can be improved to not have it.